cache/eventsource.go

```diff
@@ -0,0 +1,77 @@
+package cache
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	rpctypes "go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+type eventSource struct {
+	cli       *clientv3.Client
```

Comment:
Short variable names should only be used in local context when they are obvious. What `cli` stands here for? Command Line Interface? Just `client` is enough. 

Same for `ch` below. Maybe `watch`?